### ISSUE TITLE
Break up the TestGrid config.yaml

### ISSUE THREAD

**Issue by michelle192837** [kubernetes/test-infra#10997]
The TestGrid config.yaml (https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml) is currently one monolithic file that can be hard to parse through (and takes a long time to load/scroll through). Instead, allow users to define separate config.yaml files (similar to Prow jobs) that are scoped to dashboards/dashboard groups, and make sure config translation from YAML to proto can read multiple files.

/area testgrid
/help

**Comment by krzyzacy** [comment]
> it shouldn't just replicate TestGrid config XD

I think for a job, the annotation just need to indicate which dashboard it need to be on if they don't need to customize more fields? (We cover most of the goodies in our defaults)

**Comment by michelle192837** [comment]
Description is the other one I can think of that most users should customize. (Possibly alerting too but I don't think it's that common). Possibly dashboard group as well (coverable by 'which sig owns this'?).

Does an annotation in Prow also mean the config is a TestGrid config, or that the config may be sometimes in Prow config, sometimes in TestGrid?

**Comment by krzyzacy** [comment]
Description can be a useful thing to share between Prow and Testgrid.

Annotation is just a generic key:value pair - maybe in the future it's also useful to link from Prow to Testgrid? xD (Also how we did that with gubernator?)

**Comment by michelle192837** [comment]
👍 

Yeah, that does seem super-useful. I have no idea how Gubernator did it XD

**Comment by ixdy** [comment]
Yes, you can use multiple YAMLs:
https://github.com/kubernetes/test-infra/blob/6cda085221f72c6f4b519de471e93f2bac610277/testgrid/cmd/configurator/main.go#L63

They're processed in the order provided, so if one provides defaults, later files will use those same defaults unless overridden.

**Comment by spiffxp** [comment]
I have been playing around with this in the background FWIW, somehow I didn't notice the multiple YAML files thing 🤦‍♂️

**Comment by ixdy** [comment]
well, it was [added back in 2017](https://github.com/kubernetes/test-infra/pull/2060), so ... :)

**Comment by michelle192837** [comment]
/remove-lifecycle-rotten

@Katharine has been doing great work in this area with Prow annotations for TestGrid dashboards and generated config ^^

**Comment by spiffxp** [comment]
Welp.  I _thought_ this issue already existed, but I couldn't find it at the time.  We're working on this now, but over in https://github.com/kubernetes/test-infra/issues/13541

/close
closing this in favor of that
